[Site] new facet menu demo WIP

Nayte91 opened the pull request:
| Q              | A
| -------------- | ---
| Bug fix?       | no
| New feature?   | no
| Deprecations?  | no
| Documentation? | yes
| License        | MIT

## WHAT
New demo where we use Live Components to address a common web use case: a facet menu to filter articles, and another Live Component-powered results section where articles are filtered in real time. Bonus: the URL is also updated in real time when facet filters change, ensuring a correct shareable link for the user.

## WHY
I believe this is a common use case, and using Live Components for it is relevant. Providing real solutions for real problems on the website offers a lot of value.

## DRAFT
Still to do:
1. Ensure the design is the best possible
2. Use the latest Live Component URL features for a top-notch implementation
3. Add more filters for animalz?

Nayte91 committed:
[Site] New facet menu demo WIP

Nayte91 committed:
[Minor][Site] typo on description

94noni commented on ux.symfony.com/src/Twig/Components/Animalz/Results.php:
```diff
@@ -0,0 +1,61 @@
+    ) {
+    }
+
+    #[LiveListener('facetSetted')]
```
```suggestion
    #[LiveListener('Animalz.FacetMenu.facetSetted')]
```

as per above suggestion if accepted

94noni commented on ux.symfony.com/src/Enum/AnimalzType.php:
```diff
@@ -0,0 +1,44 @@
+
+enum AnimalzType: string
+{
+    case Normal = 'NORMAL';
```
```suggestion
    case Normal = 'Normal';
```

wdyt?

94noni commented on ux.symfony.com/src/Service/AnimalzRepository.php:
```diff
@@ -0,0 +1,100 @@
+
+        return array_map(
+            fn (array $item): Animalz => new Animalz(
+                $item['name'],
```
```suggestion
                name: $item['name'],
```

and same for other props, wdyt?

94noni commented on ux.symfony.com/src/Twig/Components/Animalz/FacetMenu.php:
```diff
@@ -0,0 +1,90 @@
+    public function emitChange(): void
+    {
+        $this->nullTheProps();
+        $this->emit('facetSetted', $this->convertPropertiesToAssociativeArray());
```
```suggestion
        $this->emit('Animalz.FacetMenu.facetSetted', $this->convertPropertiesToAssociativeArray());
```

perhaps be more precise to locate/listen particular event ?

94noni commented on ux.symfony.com/templates/components/Animalz/Results.html.twig:
```diff
@@ -0,0 +1,21 @@
+<section {{ attributes }} class="card">
+    <header class="card-header d-flex justify-content-between align-items-center">
+        <span>Results</span>
+        <span class="badge bg-secondary">{{ this.results|length }}</span>
```
is it good practice to call twice `this.results` ?

here and in the loop below

94noni reviewed (commented):
some review passing by
I think its a good addition as you mentioned in the PR doc :)

Nayte91 committed:
[Site] Address PR review feedback

- Use PascalCase for enum values (Normal instead of NORMAL)
- Use named arguments in Animalz constructor
- Cache this.results in template variable to avoid double call

smnandre commented:
@Nayte91 i'm not sure how to do this now... should i push the changes I made directly there ? 
... or I think I can open a PR on your PR for you to see them and merge (or not) in your branch

(disclosure: we discussed this PR on Slack for some time with Nayte and I started the other day some changes / moving things around)

TL;DR;  some code clean, simplification, pushed your design a bit in the spirit of what we have already, and anticipate the fact this demo would benefit from alternative choices... 

---

Facet search, "grid of products" is probably what I'm asked the most, about Symfony UX... 

And I can say there is no "one-for-all" solution. 

I really do like your demo and we will happily publish it. 
But after\* that I will publish two other ways / alternative "solutions", that are also relevant (even if one is a bit .. original 😅 ) Point is I would not like users to feel "ho, this is THE way".   This is why i'm trying to anticipate the reuse of HTML/components, a bit as I did with the scroll. 

(but you know how I'm really not able to say right now if "after that" is in two days or two years...)

Nayte91 committed:
Merge branch 'symfony:2.x' into docs/demo-facet-menu

smnandre committed:
sa: rework demo

94noni reviewed (approved):
TIL few stuff reading here 👍🏻

smnandre commented:
What's next here @Nayte91 ? Are you expecting something from my side ?

Kocal commented:
Hey, just to let you know, the website has moved to an external repository https://github.com/symfony/ux.symfony.com so there is no point to keep this PR open here.

Thanks for your contribution, and do not hesitate to re-open the PR on website repository!
